### PR TITLE
Pull request related to issue #1092 and #1093

### PR DESCRIPTION
**Pull request by jvalegre**
Pull request related to issue #1092. The code broke due to a problem related to the mini print level of ORCA. 

Changes:
1. When parsing ORCA files using the mini print level, cclib raises a warning instead of breaking. The code prints a warning about this, but ideally the users should avoid the mini print level. For example, multiplicity just gets assigned to 1 since it's not included at this print level. The code doesn't break when using the mini level but the fix isn't ideal, since I set rmsDP_value and rmsDP_target to 0.0. 
2.  Final single point E included. This is the final energy we use for single-point energy corrections in GoodVibes, but it wasn't included (I think).

I'm attaching the output files from ORCA and the json files generated with ccwrite, please let me know if you need more information!

[TESTS.zip](https://github.com/cclib/cclib/files/8023283/TESTS.zip)

----------
Pull request related to issue #1093. There was a bug when retrieving the NBO (natural) charges for open-shell systems, since the code retrieved only the charges from the b-density rather than the total density. The code also includes new parameters in the json files generated with ccwrite from Gaussian calculations that GoodVibes and AQME use (already commented to @berquist):

1. Parameters from NMR calcs
image

2. Metadata
image

3. Symmetry point group, rotational constants and T
image

4. Information about <S**2> before and after spin annihilation
image

5. Corrected natural (NBO) charges and spins
image

6. Wiberg bond order matrix (from NBO)
image

----------
Other fixes and additions:
1. Thermochemistry (zpe, enthalpy, free energy) from Gaussian outputs was in hartree, now in eV
2. Time dependent (TD) energies added from Gaussian outputs
3. ONOIM energies added from Gaussian outputs
4. G4 energies added from Gaussian outputs
5. Final single-point energy added from ORCA outputs

### CODE REVIEW

**Comment by jvalegre** [comment]
hi @berquist , hope everything is going well! 

I'd like to include similar data for ORCA and NWChem (I only added the GoodVibes parameters for Gaussian). Did you start working on this pull request or I can make a couple more changes to my local cclib code?

Thanks!

**Review by berquist — CHANGES_REQUESTED** [review]
The most important comment I can give is the one about breaking up this PR into smaller ones based on bug fixes and feature additions.

For the new attributes, https://github.com/cclib/cclib/issues/1093#issuecomment-1018112286 and the documentation were not clear enough about
- what should be placed as attributes on `ccData`,
- what should be a calculation method (under `cclib/methods`), and
- what should be placed in metadata (`ccData.metadata`, a string to _any type_ dict, https://cclib.readthedocs.io/en/latest/data_notes.html#metadata).

I see you put a bunch of stuff under the JSON path for `metadata`.  This is good, but makes me realize that there is no guiding documentation about what to put there, and also that `metadata` is actually an attribute itself (the one linked above).  Rather than let this review sit even longer, though we'll want to change a bunch of locations of things, can you split this PR apart first?

File: cclib/parser/orcaparser.py [hunk 1]
```diff
@@ -150,6 +150,8 @@ def extract(self, inputfile, line):
                 if line[0] != '|':
                     break
                 lines.append(line[line.find('> ')+2:])
+                if 'printlevel mini' in line or 'miniprint' in line:
+                    print("WARNING: print level is set to mini, which might cause the loss of relevant information")
```
[the remark this answers is not in the document]
Comment (jvalegre) [reply cclib/parser/orcaparser.py:154]: done

File: cclib/parser/gaussianparser.py [hunk 2]
```diff
@@ -2090,11 +2313,14 @@ def extract_charges_spins(line,prop):
         #Sum of electronic and thermal Enthalpies=            -563.635755
         #Sum of electronic and thermal Free Energies=         -563.689037
         if "Zero-point correction" in line:
-            self.set_attribute('zpve', float(line.split()[2]))
+            zpve = utils.convertor(utils.float(line.split()[2]), "hartree", "eV")
+            self.set_attribute('zpve', zpve)
```
[the remark this answers is not in the document]
Comment (jvalegre) [reply cclib/parser/gaussianparser.py:2327]: thanks!

File: cclib/parser/gaussianparser.py [hunk 3]
```diff
@@ -2074,10 +2291,16 @@ def extract_charges_spins(line,prop):
                 if line1.split()[0] == 'Natural' and line2.split()[2] == 'Charge':
                     dashes = next(inputfile)
                     charges = []
+                    spins = []
                     for i in range(self.natom):
                         nline = next(inputfile)
                         charges.append(float(nline.split()[2]))
+                        spins.append(float(nline.split()[-1]))
                     self.atomcharges["natural"] = charges
+                    try:
```
[the remark this answers is not in the document]
Comment (jvalegre) [reply cclib/parser/gaussianparser.py:2310]: I refactored it, however, this code comes from a previous version that doesn't separate Mulliken & NBO charges and spins (i.e. NBO charges and spins just override Mulliken values). Should I separate charges and spins into charges Mulliken, charges NBO, spins Mulliken and spins NBO?

File: cclib/parser/orcaparser.py [hunk 4]
```diff
@@ -778,7 +801,7 @@ def splitter(line):
             self.mosyms = [[]]
 
             line = next(inputfile)
-            while len(line) > 20:  # restricted calcs are terminated by ------
+            while len(line) > 20 and not 'Total SCF time' in line:  # restricted calcs are terminated by ------
```
[the remark this answers is not in the document]
Comment (jvalegre) [reply cclib/parser/orcaparser.py:804]: yeah, let me distribute the fixes/additions into Gaussian and ORCA

File: cclib/parser/gaussianparser.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (berquist) [reply cclib/parser/gaussianparser.py]: > Should I separate charges and spins into charges Mulliken, charges NBO, spins Mulliken and spins NBO?

Yes please! This must have been an oversight and I can definitely see people would want clearly identified NBO properties. This means parsed output of any NBO calculation right now is incorrect.

**Pull request closed without merging**